Adds collections integrity unit tests

ggirotto opened the pull request:
Also adds updated - added `resources` and made other small adjustments - collections to `collections/` folder.

matuella commented on firebase/test/collections/integrity-check.spec.ts:
```diff
@@ -1,6 +1,61 @@
+    rawLocalCollections = localCollections.map((collection) => JSON.parse(collection));
+  });
+
+  it("should have at least one collection", () => assert.ok(rawLocalCollections.length > 0));
```
Why should it? I Don't think it's an useful test

matuella commented on firebase/test/collections/integrity-check.spec.ts:
```diff
@@ -1,6 +1,61 @@
+    });
+  });
+
+  it("should have unique memo ids amongst themselves", () => {
```
```suggestion
  it("should have unique memo ids in each collection", () => {
```

Why should they have a unique global id?

ggirotto replied:
I copied this test from the client, I though it was a definition, but we may remove this test

matuella commented on firebase/test/collections/integrity-check.spec.ts:
```diff
@@ -1,6 +1,61 @@
+  });
+
+  it("should have expected JSON structure", () => {
+    const schemaValidator = Provider.instance.schemaValidator;
```
it seems odd to use provider here, why not create an instance of the expected schema validator?

matuella reviewed (commented):
Shouldn't we move these tests to integration tests of some sort?

matuella commented:
Also, I've already removed the test collection in `https://github.com/olmps/memo/pull/276`. Will merge both #275 and #276 so you can rebase on that.

matuella reviewed (commented):
Will these collections properly run at the same time that the workflow is merged into `main`?

Like, we're developing in `server`, then, when merging into `main`, will the `.github/` changes work effectively with the script itself?

ggirotto commented:
> Will these collections properly run at the same time that the workflow is merged into `main`?
> 
> Like, we're developing in `server`, then, when merging into `main`, will the `.github/` changes work effectively with the script itself?

Good catch. Updated `update-collections` workflow to run the integrity tests before running the collections sync.

matuella commented:
> Good catch. Updated update-collections workflow to run the integrity tests before running the collections sync.

But does it change the fact that all collections are going to be merged before the workflow is available?

ggirotto commented:
> But does it change the fact that all collections are going to be merged before the workflow is available?

I didn't understood your question. The `update-collections` is the one who merges the collections in Firestore. If it fails in the tests lane, it doesn't continue with the merging process.

matuella commented:
This workflow will only work when changes are pushed to `main` branch. All collections here in this pull will be available at the same time as the workflow in the `main` branch, which, in my naive perspective, wouldn't run the sync, as it will only detect such changes, in that particular folder, after it has been available in `main`.

TLDR: shouldn't we split merging the collections after the server is available in main?

ggirotto commented:
Actually I believe they'll be synched, because the workflow runs on every push on main detecting the changes since the last merge commit, which includes all added collections. It'll compare the merge commit in `main` with [b15d97](https://github.com/olmps/memo/commit/b15d976f7b7830cee7cac48623ab4dbf096b910c), which is the last commit, and in this difference these added collections will be considered.

matuella reviewed (approved):
Then I believe we should test it and see if it working as expected, otherwise split the `server` merge into the actual changes and the collections to-be-added.

I'll aprove this nonetheless, but please keep this is mind.
